OmniKit/PumpManager/PodState.swift

```diff
@@ -196,10 +196,14 @@ public struct PodState: RawRepresentable, Equatable, CustomDebugStringConvertibl
         }
     }
     
-    private mutating func updateDeliveryStatus(deliveryStatus: DeliveryStatus) {
+    private mutating func updateDeliveryStatus(deliveryStatus: DeliveryStatus, podProgressStatus: PodProgressStatus, bolusNotDelivered: Double) {
         finalizeFinishedDoses()
 
-        if let bolus = unfinalizedBolus, bolus.scheduledCertainty == .uncertain {
+        // Add the unfinalizedBolus here if there isn't one and are currently bolusing in a ready state.
+        if unfinalizedBolus == nil && deliveryStatus.bolusing && podProgressStatus.readyForDelivery {
```

Comment:
Have you actually observed a state like this outside of killing Xcode?

Reply:
No, but having like this is critical for this PR. If a pod ever gets a bolus command when a bolus is currently in progress, the pod will immediately fail with an $31 fault. The unfinalizedBolus is what keeps track of in-progress boluses, so it's critical to have this correctly set up for any in-progress bolus. I used an Xcode breakpoint to simulate a user stopping Loop at the exact wrong time right after the bolus has started, but before the unfinalizedBolus is created. Without having this code, Loop would fault the pod if an automatic bolus was attempted on Loop restart if the bolus was still on going if the app was stopped at the exact spot. Having this code also protects against a pod fault if Loop ever has some logic error where an unfinializedBolus is never initialized or it is incorrectly removed/deleted before the bolus has actually completed.

In summary, implementing a partial recovery of an unrecorded/lost bolus isn't actually a separate thing, but it is required to avoid $31 pod faults when not doing a get status call before each bolus.

Reply:
This seems like unlikely to be used code, that makes understanding what's going on here more difficult than it needs to be. I really dislike this pattern of defensive coding for problems that haven't been observed. It seems much more likely that the pod might take longer to deliver a bolus than we expect, which is why we had the status check in the first place.

Reply:
Rewrote things a bit to simplify this part. This code is needed to handle a bolus being delivered and then the user moving out of range (before the response is received & the subsequent getStatus completes trying to verify the bolus) and then the user stops Loop. There is new logic in enactBolus() to do a getStatus() if enough time has passed for the bolus to finish, but not enough time for Loop to be absolutely positive.